app/models/live_metric.rb

```diff
@@ -3,6 +3,10 @@ class LiveMetric < ActsAsArModel
 
   class LiveMetricError < RuntimeError; end
 
+  def self.virtual_attribute?(c)
+    super || c =~ /^mw_/
```

Comment:
I think return true could be just enough.
Basically, on LiveMetric.find(), this is resolved, so option 2 would need to store additional state at class level.
So, at this use case has some boundaries I do not see any risk in additional use cases.
LiveMetric is mainly used for miq_reports, so if there is some wrong column, I think there will be an error on log which should be expected.